sdk/src/main/java/io/radar/sdk/RadarApiHelper.kt

```diff
@@ -85,6 +80,33 @@ internal open class RadarApiHelper(
                 }
 
                 if (params != null) {
+                    val prevUpdatedAtMsDiff = params.optLong("updatedAtMsDiff", -1L)
+                    val replays = params.optJSONArray("replays")
+
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1 && (prevUpdatedAtMsDiff != -1L || replays != null)) {
+                        val nowMs = SystemClock.elapsedRealtimeNanos() / 1000000
+                        val locationMs = params.optLong("locationMs", -1L)
+                        if (prevUpdatedAtMsDiff != 1L && locationMs != -1L){
```

Comment:
So I get updating the replays in the code below, but why do we need to update the regular part of the request - haven't we already calculated that in `RadarApiClient `line 307?

Reply:
This is to correctly time a `/track` call that's been client-side delayed by the `request(..., sleep: true)`. If there's two tracks at t0 and t0.1, the second track at t0.1 will have it's request happen at `1s + request latency of t0`.

For the server to compute the correct timestamp of the second `/track` as 0.1 seconds after the first, it will need to have that larger `updatedAtMsDiff` that comes from recalculating here.